promql/engine.go

```diff
@@ -728,16 +762,25 @@ func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ..
 			ev.errorf("vector cannot contain metrics with the same labelset")
 		}
 		enh.out = result[:0] // Reuse result vector.
+
+		// In some cases we only have the result of 'f' without having tracked any samples
+		// in the above for loop that built the input vectors.
+		ev.currentSamples += len(result)
```

Comment:
You want to add this to tempNumSamples as the output data from previous cycles is in memory.

Reply:
Hmmm, sorry I don't understand why. 

In addition to, or instead of adding it to `ev.currentSamples`? It seems to me that it should be in addition to since we check `ev.currentSamples` when determining if we have too many samples in memory.

Reply:
Yes. After every iteration we're now holding the results of previous iterations in memory, which needs to be part of tempNumSamples.